products/bridge/test/test.ts

```diff
@@ -396,17 +416,15 @@ describe("Bridge", function () {
       success,
       result,
       nonce,
-      signerIndices,
       signatures
     );
 
-    const blockNum = await ethers.provider.getBlockNumber();
     const filter = twin2.filters.Succeeded;
-    const logs = await twin2.queryFilter(filter, blockNum - 100, blockNum);
-    const res = logs[0].args[0];
-    expect(res).to.equal(num + 1);
+    const logs = await twin2.queryFilter(filter);
```

Comment:
As discussed, it's essential to only consider events in finalized blocks, therefore the code before refactoring fetched `ethers.provider.getBlockNumber('finalized')` and queried only until that block number.